modulus/utils/graphcast/graph.py

```diff
@@ -102,9 +125,20 @@ def create_mesh_graph(self, verbose: bool = True) -> Tensor:
         # ensure fields set to dtype to avoid later conversions
         mesh_graph.ndata["x"] = mesh_graph.ndata["x"].to(dtype=self.dtype)
         mesh_graph.edata["x"] = mesh_graph.edata["x"].to(dtype=self.dtype)
+        if self.khop_neighbors > 0:
+            # Make a graph whose edges connect the k-hop neighbors of the original graph.
+            mesh_graph = dgl.khop_graph(
```

Comment:
Note: This is not very efficient and we need to also look for alternative approaches. Takes a long time to compute the khop graph of a level 6 mesh.

Reply:
My bad, I forgot that khop_graph doesn't "filter out" duplicate edges between nodes but actually just aggregates all different k-hop graphs. This is not what we need and leads to a lot of unnecessary edges. I can imagine this being slow. 
Also: the way through networkx is also likely not ideal.

DGL also has `khop_adj` which is actually what we want, i.e. it returns the adjancency matrix of A^k. Can you try
something like `sum([dgl.khop_adj(g, k=k) for k in range(1, kmax+1)]).bool().float()` which directly should give you a binary adjacency matrix?

Reply:
(you later have to invert the binary matrix and scale it accordingly for a valid mask)

Reply:
Looked a bit into the internal of khop_adj, by avoiding the backend-agnostic nature of DGL, some unnecessary conversions between torch, numpy, and scipy-sparse, it actually is somewhat easy to implement a standalone function relying only on the SparseTensor of PyTorch itself.

```
def khop_adj_all_k(g, kmax):
    assert g.is_homogeneous, \
        'only homogeneous graph is supported'
    with torch.no_grad():
        adj = g.adj_external(transpose=True, scipy_fmt=None)
        adj_k = adj.clone()
        adj_all = adj.clone()
        for _ in range(2, kmax + 1):
            adj_k = adj_k @ adj
            adj_all += adj_k

    return adj_all.to_dense()
```

This should return the union of all k-hop subgraphs for k in (1, ..., kmax) and surely runs a lot faster.